external-builds/pytorch/skip_tests/create_skip_tests.py

```diff
@@ -0,0 +1,252 @@
+#!/usr/bin/env python3
+"""PyTorch Test Skip List Generator.
+
+Creates a list of tests to be skipped or the only ones to be run on for PyTorch's pytest.
+
+Key Features
+------------
+- AMDGPU family-specific test filtering
+- PyTorch version-specific test filtering
+- Can generate both skip lists and include lists
+- Supports standalone CLI usage and programmatic API
+
+Module Structure
+----------------
+The module expects skip test definitions in the following files:
+- generic.py: Common tests to skip across all configurations
+- pytorch_<version>.py: Version-specific tests (e.g., pytorch_2.7.py)
+
+For additional information have a look at the README.md file in this directory.
+
+Usage Examples
+--------------
+Programmatic usage:
+    # Create a list of tests to be skipped as they are known to be failing on gfx942 and PyTorch 2.7
+    from create_skip_tests import get_tests
+    skip_expr = get_tests(["gfx942"], "2.7", create_skip_list=True)
+    # Use skip_expr with pytest: pytest -k "{skip_expr}"
+
+    # Skip tests for multiple GPU families
+    skip_expr = get_tests(["gfx1011", "gfx1012"], "2.7", create_skip_list=True)
+
+Command-line usage:
+    Run all test excluding known failures for gfx942 and PyTorch 2.7:
+    $ python create_skip_tests.py --amdgpu-family gfx942 --pytorch-version 2.7
+
+    Run tests excluding failures for multiple GPU families:
+    $ python create_skip_tests.py --amdgpu-family "gfx1011,gfx1012" --pytorch-version 2.7
+
+    Run all tests that are normally skipped for gfx942 and all pytorch versions:
+    $ python create_skip_tests.py --amdgpu-family gfx942 --pytorch-version all --include-tests
+
+"""
+
+import argparse
+import glob
+import importlib.util
+import os
+import sys
+from typing import Dict, List
+
+
+def import_skip_tests(pytorch_version: str = "") -> Dict[str, Dict]:
+    """Dynamically load test skip definitions from configuration files.
+
+    Loads skip test definitions from:
+    - generic.py (always loaded)
+    - pytorch_<version>.py (if version specified)
+    - pytorch_*.py (all version files if version="all")
+
+    Args:
+        pytorch_version: PyTorch version string (e.g., "2.7", "all", or "").
+            - "" (empty): Load only generic.py
+            - "all": Load generic.py and all pytorch_*.py files
+            - Specific version: Load generic.py and pytorch_<version>.py
+
+    Returns:
+        Dictionary mapping module names to their skip_tests dictionaries.
+        Format: {"generic": {...}, "pytorch_2.7": {...}}
+
+    """
+    this_script_dir = os.path.dirname(os.path.abspath(__file__))
+
+    files = [os.path.join(this_script_dir, "generic.py")]
+    if pytorch_version == "all":
+        files += glob.glob(os.path.join(this_script_dir, "pytorch_*.py"))
+    elif pytorch_version:
+        files += [os.path.join(this_script_dir, f"pytorch_{pytorch_version}.py")]
```

Comment:
nit: could use pathlib here (though this usage is relatively simple and doesn't contain any obvious portability issues)